Doc/whatsnew/3.12.rst

```diff
@@ -102,6 +102,29 @@ Deprecated
 Removed
 =======
 
+* Removed many old deprecated :mod:`unittest` features:
```

Comment:
We should replicate that deprecated aliases table from unittest.rst in here, inverted so that we show a mapping of the old deprecated names -> the single now required name.